### PR TITLE
EVG-6879 / EVG-6900 update schema for historical test stats

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

File: model/testresult/testresult.go [hunk 1]
```diff
@@ -17,22 +19,32 @@ const (
...
+
+	// Denormalize TestResults to add fields from enclosing tasks. This
+	// facilitates faster historical test stats  aggregations.
+	Project              string    `bson:"branch" json:"branch"`
+	BuildVariant         string    `bson:"build_variant" json:"build_variant"`
+	DistroId             string    `bson:"distro" json:"distro"`
+	Requester            string    `bson:"r" json:"r"`
+	DisplayName          string    `bson:"display_name" json:"display_name"`
+	ExecutionDisplayName string    `bson:"execution_display_name,omitempty" json:"execution_display_name,omitempty"`
+	CreateTime           time.Time `bson:"create_time" json:"create_time"`
```
Comment (tychoish) [inline model/testresult/testresult.go]: can we also have a complete time so we can migrate away from the floats?

Comment (jimoleary) [reply model/testresult/testresult.go]: @tychoish 
I think that this change request is out of scope for this work.

The start time and end time will continue to be used regardless.

Comment (jimoleary) [reply model/testresult/testresult.go]: It's completely trivial, but on the other hand it is also completely unrelated to this fix.

Comment (tychoish) [reply model/testresult/testresult.go]: Please do it anyway. It doesn't make sense to change the schema multiple times.

Comment (tychoish) [reply model/testresult/testresult.go]: The Start/End Fields should have time.Time equivalents in the new schema.

**Comment by jimoleary** [comment]
Updated, I called the collection fields start_date and end_date just so that it is not confusing.

**Comment by jimoleary** [comment]
create_time is the task_create_time. start_date is the test_start_time.

StartTime (start in the database) and EndTime (end) need to remain the same as they are already in use.

But I can change start_date, end_data and create_time to make them nore obvious. 

AFAICS the failing tests are unrelated to these changes but I'll look into it.

**Comment by jimoleary** [comment]
Rebased onto master and the tests that now fail have changed. There is a failure due to auth in test-rest-route and a setup failure in test-cloud, neither of which are related to this set of changes.

@tychoish I've changed the struct field names (that can be changed). Hopefully they are more clear now.

**Pull request merged**